### PR TITLE
Install and configure eslint [Resolves #72]

### LINKED ISSUE #72: Setup ES6 Lint Preferences to align with JRob's Specifications

### PR DESCRIPTION
**Pull request by jason00111**
- fix linting errors
- add "lint-fix" script to package.json

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +27,7 @@
     "webpack": "^1.13.3"
   },
   "scripts": {
-    "start": "react-scripts start",
+    "start": "eslint src && react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test --env=jsdom",
     "eject": "react-scripts eject"
```
Comment (AbrahamFergie) [inline package.json]: Looks good to me

File: package.json [hunk 2]
```diff
@@ -21,9 +27,10 @@
     "webpack": "^1.13.3"
   },
   "scripts": {
-    "start": "react-scripts start",
+    "start": "eslint src && react-scripts start",
```
Comment (jrob8577) [inline package.json]: We don't want to do this because this means run eslint, and if it succeeds, run react-scripts start.  I think we need a separate lint watch command.

File: src/actions/widgets/index.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 export const CLOCK_TICK = 'clockTick'
 
-const clockTick = time => ({ type: CLOCK_TICK, time })
+const clockTick = time => ( { type: CLOCK_TICK, time } )
```
Comment (jrob8577) [inline src/actions/widgets/index.js]: Ew. Let's add an exception to the [space-in-parens](http://eslint.org/docs/rules/space-in-parens) rule for curly braces.
```
{ "exceptions": ["{}"] }
```

File: src/components/AccountMenuDropdown.js [hunk 4]
```diff
@@ -12,9 +12,9 @@ export default class AccountMenuDropdown extends Component {
   }
 
   toggleMenu() {
-    this.setState({
+    this.setState( {
```
Comment (jrob8577) [inline src/components/AccountMenuDropdown.js]: Same as above, curly braces should be an exception to the spaces in parens.

File: src/components/Dashboard.js [hunk 5]
```diff
@@ -18,4 +18,4 @@ class Dashboard extends Component {
   }
 }
 
-export default connect( state => ({ widgets: state.widgets }))( Dashboard )
+export default connect( state => ( { widgets: state.widgets } ) )( Dashboard )
```
Comment (jrob8577) [inline src/components/Dashboard.js]: :|

File: src/components/DashboardBody.js [hunk 6]
```diff
@@ -12,7 +12,7 @@ const buildWidgets = widgets =>
       'Image': <Image { ...widget } key={ widget.title } />,
       'QRCode': <QRCode { ...widget } key={ widget.title } />,
       'Clock': <Clock { ...widget } key={ widget.title } />
-    }[ widget.type ]))
+    }[ widget.type ] ) )
```
Comment (jrob8577) [inline src/components/DashboardBody.js]: This looks funny as well, seems like we need another exception for square brackets.

File: src/reducers/widgets/index.js [hunk 7]
```diff
@@ -11,17 +11,17 @@ const updateWidget = ( state, action, type, updater ) => {
...
   ]
 }
 
 const widgetReducer = ( state = [dummyClock], action ) => {
   switch( action.type ) {
-    case CLOCK_TICK:
-      return updateWidget( state, action, 'Clock', clockTicked )
-    default:
-      return state
+  case CLOCK_TICK:
```
Comment (jrob8577) [inline src/reducers/widgets/index.js]: I prefer case statements indented under their enclosing switch - can we find the appropriate linting rule for that? (Or just undo this change)

File: src/components/dashboard/widgets/Image.js [hunk 8]
```diff
@@ -12,7 +12,10 @@ export default class Image extends Component {
 
     return (
       <article>
-        <div className="widget widget-width-1 widget-height-1 transparent" style={ positionStyle }>
+        <div
+          className="widget widget-width-1 widget-height-1 transparent"
+          style={ positionStyle }
+        >
```
Comment (jrob8577) [inline src/components/dashboard/widgets/Image.js]: HTML/JSX tags should be closed on the same line as their last attribute

File: package.json [hunk 9]
```diff
@@ -4,6 +4,11 @@
   "private": true,
   "devDependencies": {
     "enzyme": "^2.6.0",
+    "eslint": "^3.12.2",
+    "eslint-plugin-import": "^2.2.0",
+    "eslint-plugin-jsx-a11y": "^3.0.2",
```
Comment (jrob8577) [inline package.json]: I don't see this getting used in the linting rules - what's its purpose?

File: src/components/dashboard/widgets/QRCode.js [hunk 10]
```diff
@@ -12,7 +12,10 @@ export default class QRCode extends Component {
 
     return (
       <article>
-        <div className="widget widget-width-1 widget-height-1" style={ positionStyle }>
+        <div
+          className="widget widget-width-1 widget-height-1"
+          style={ positionStyle }
+        >
```
Comment (jrob8577) [inline src/components/dashboard/widgets/QRCode.js]: HTML/JSX tags should be closed on the same line as their last attribute

File: src/components/dashboard/widgets/Text.js [hunk 11]
```diff
@@ -11,7 +11,10 @@ export default class Text extends Component {
 
     return (
       <article>
-        <div className="widget widget-width-1 widget-height-1" style={ positionStyle }>
+        <div
+          className="widget widget-width-1 widget-height-1"
+          style={ positionStyle }
+        >
```
Comment (jrob8577) [inline src/components/dashboard/widgets/Text.js]: HTML/JSX tags should be closed on the same line as their last attribute

File: src/components/landing/Body.js [hunk 12]
```diff
@@ -9,16 +9,18 @@ import '../../styles/landing/body.css'
...
+        <section className="quotations">
+          <div className="quotations-container">
+            <Quotation quotation="I found an apple thanks to Dragonboard"
+              url="/images/gravity2.png"
+              name="Isaac Newton" />
+            <Quotation
+              quotation="We installed Dragonboard, sales are now up by 15%"
+              url="images/how-to-draw-john-adams-john-adams_1_000000016809_5.png"
+              name="John Adamos, CEO, Adam Apples"
+            />
```
Comment (jrob8577) [inline src/components/landing/Body.js]: HTML/JSX tags should be closed on the same line as their last attribute (especially odd here since it's not internally consistent with the `Quote`, above)

File: src/components/landing/navigation/Navigation.js [hunk 13]
```diff
@@ -9,29 +9,33 @@ import '../../../styles/landing/navigation/menu.css'
...
-              <a className='demonstration-button' href='#'>BOOK A DEMO</a>
-              <input className='trial-input' type='text' placeholder='Email address' />
-              <a className='navigation-trial-button' href='#'>Free Trial</a>
+              <a className="login-navigation" href="#">LOGIN</a>
+              <a className="demonstration-button" href="#">BOOK A DEMO</a>
+              <input
+                className="trial-input"
+                type="text"
+                placeholder="Email address"
+              />
```
Comment (jrob8577) [inline src/components/landing/navigation/Navigation.js]: HTML/JSX tags should be closed on the same line as their last attribute

File: package.json (hunk 9 above)
Comment (jason00111) [inline package.json]: It was doing nothing, we removed it
